Diagnostics/AVTester/Test-ExchAVExclusions.ps1

```diff
@@ -104,205 +206,432 @@ if (-not($exchangeShell.ShellLoaded)) {
+
+        # Look at if we have found it
+        if ($null -eq $RunningProcess) {
+            Write-SimpleLogFile -string "Process $Process not found" -name $LogFile
         } else {
+            Write-SimpleLogFile -string "Found $Process" -name $LogFile
+
+            $runningProcessesCounter = 0
+            # Pull each instance of the process
+            foreach ($Instance in $RunningProcess) {
+                Write-Progress -Id 1 -ParentId 0 -Activity "Examining $Process $($runningProcessesCounter+1) of $($RunningProcess.Count)" -Status " "
+
+                if ( $Process -eq $Instance.Path) {
+                    Write-SimpleLogFile -String ("############################################################") -name $LogFile
+                    Write-SimpleLogFile -String ("Working on Process $($Instance.Path)") -name $LogFile
+                    Write-SimpleLogFile -String ("Working on PID $($Instance.Id)") -name $LogFile
+                    $CommandLine = (Get-CimInstance  -Query "Select * from Win32_Process where ProcessId = '$($Instance.Id)'" | Select-Object CommandLine).CommandLine
+
+                    Write-SimpleLogFile -String ("CommandLine: $CommandLine") -name $LogFile
+
+                    $instancesCounter = 0
+                    foreach ($module in $Instance.Modules ) {
+                        Write-Progress -Id 2 -ParentId 1 -Activity "Examining loaded modules in $($Instance.Id) process." -PercentComplete (($instancesCounter / $Instance.Modules.Count) * 100) -Status " "
+                        Write-SimpleLogFile -String ("Working on Module $($module.FileName)") -name $LogFile
+                        $signature = $module.FileName | Get-AuthenticodeSignature
+                        if ( $signature.Status -eq 'NotSigned') {
+                            if ( ($module.FileName.ToLower().StartsWith(('c:\windows\assembly\NativeImages_').ToLower()) -and (
+                                        $module.FileName.ToLower().EndsWith('.ni.dll') -or $module.FileName.ToLower().EndsWith('.ni.exe') -or
+                                        $module.FileName.ToLower().EndsWith('.wrapper.dll') ) ) -or
+                            ($module.FileName.ToLower().StartsWith('c:\windows\system32\') -and [environment]::OSVersion.Version.Major -le 6 ) -or
+                            ($module.FileName.ToLower().StartsWith(('c:\windows\WinSxS\').ToLower()) -and [environment]::OSVersion.Version.Major -le 6 ) -or
+                            ($module.FileName.ToLower().StartsWith('c:\windows\assembly\') -and [environment]::OSVersion.Version.Major -le 6 ) -or
+                            ($module.FileName.ToLower().StartsWith('c:\windows\microsoft.net\assembly\gac_64\') -and [environment]::OSVersion.Version.Major -le 6 ) -or
+                            ($module.FileName.ToLower() -eq (Join-Path ($env:ExchangeInstallPath).ToLower() 'bin\osafehtm.dll') -and ([byte]$serverExchangeInstallDirectory.MsiProductMinor) -gt 0 ) ) {
```

Comment:
Add comment for what we are trying to do here. This is extremely complicated and need more context here. But I believe this is testing for things that aren't signed for known modules that are on Windows 2012 R2 or below. 

We should just do a list of known unsigned files if it isn't long. As this would be better for security reasons.